sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandler.java

```diff
@@ -0,0 +1,599 @@
+    }
+
+    // Special printing so that durations smaller than one second
+    // are left padded with blanks instead of '0' characters.
+    // E.g.,
+    //        Number                  Printout
+    //        ---------------------------------
+    //        0.000534                  .   534
+    //        1.000534                 1.000534
+    long deltaMicros = TimeUnit.NANOSECONDS.toMicros(event.getEpochNanos() - lastEpochNanos);
+    String deltaString;
+    if (deltaMicros >= 1000000) {
+      deltaString = String.format("%.6f", (deltaMicros / 1000000.0));
+    } else {
+      deltaString = String.format("%1s.%6d", "", deltaMicros);
+    }
+
+    long microsField = TimeUnit.NANOSECONDS.toMicros(event.getEpochNanos());
+    formatter.format(
+        "%02d:%02d:%02d.%06d</pre></td> "
+            + "<td class=\"border-left-dark\"><pre class=\"no-margin wrap-text\">%s</pre></td>"
+            + "<td class=\"border-left-dark\"><pre class=\"no-margin wrap-text\">%s</pre></td>",
+        calendar.get(Calendar.HOUR_OF_DAY),
+        calendar.get(Calendar.MINUTE),
+        calendar.get(Calendar.SECOND),
+        microsField,
+        deltaString,
+        htmlEscaper().escape(renderEvent(event)));
+  }
+
+  private static String renderAttributes(ReadableAttributes attributes) {
+    final StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append("Attributes:{");
+    final AtomicBoolean first = new AtomicBoolean(true);
+    attributes.forEach(
+        new KeyValueConsumer<AttributeValue>() {
+          @Override
+          public void consume(String key, AttributeValue value) {
+            if (!first.getAndSet(false)) {
+              stringBuilder.append(", ");
+            }
+            stringBuilder.append(key);
+            stringBuilder.append("=");
+            stringBuilder.append(value.toString());
+          }
+        });
+    stringBuilder.append("}");
+    return stringBuilder.toString();
+  }
+
+  private static String renderEvent(Event event) {
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(event.getName());
+    if (!event.getAttributes().isEmpty()) {
+      stringBuilder.append(" | ");
+      stringBuilder.append(renderAttributes(event.getAttributes()));
+    }
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Emits HTML body content to the {@link PrintWriter} {@code out}. Content emited by this function
+   * should be enclosed by <body></body> tag.
+   *
+   * @param out The {@link PrintWriter} {@code out}.
+   */
+  private void emitHtmlBody(Map<String, String> queryMap, PrintWriter out)
+      throws UnsupportedEncodingException {
+    if (dataAggregator == null) {
+      out.write("OpenTelemetry implementation not available.");
+      return;
+    }
+    // Link to OpenTelemetry Logo
+    out.write(
+        "<img style=\"height: 90px;\""
+            + "src=\"https://opentelemetry.io/img/logos/opentelemetry-horizontal-color.png\" />");
```

Comment:
It would probably be better not to link to this. If it gets moved, then every zpage view in the world will be broken. Better would be to have it hosted internal to the module.

Reply:
or use a `data:image/png;base64` encoding

Reply:
Thanks for the suggestion, changed to use base64 encoding